db.go

```diff
@@ -176,35 +177,41 @@ func (db *sqlDB) PrepareContext(ctx context.Context, query string) (_stmt Stmt,
 
 // Query executes a query that returns rows, typically a SELECT.
 // The args are for any placeholder parameters in the query.
-// Query uses a radonly db as the physical db.
 func (db *sqlDB) Query(query string, args ...interface{}) (*sql.Rows, error) {
 	return db.QueryContext(context.Background(), query, args...)
 }
 
 // QueryContext executes a query that returns rows, typically a SELECT.
 // The args are for any placeholder parameters in the query.
-// QueryContext uses a radonly db as the physical db.
-func (db *sqlDB) QueryContext(ctx context.Context, query string, args ...interface{}) (*sql.Rows, error) {
-	rows, err := db.ReadOnly().QueryContext(ctx, query, args...)
-	if isDBConnectionError(err) {
+func (db *sqlDB) QueryContext(ctx context.Context, query string, args ...interface{}) (rows *sql.Rows, err error) {
+	var curDB *sql.DB
+	query = strings.ToUpper(query)
+	writeFlag := strings.Contains(query, "RETURNING")
```

Comment:
This is so specific to Postgres.
I'm still looking at the idea of how to make it optional or configurable

Reply:
I think it's better to filter based on `INSERT` `UPDATE` and `DELETE` and it's more agnostic to any kind of DB

Reply:
Ok Thanks the pr is complete